Add guidelines to enable monitoring for a Kyma component (#211)
kyma-project/community: guidelines/add-monitoring-to-component-docs.md

# Add monitoring to a Kyma component

## Enable dashboard in Grafana

> - Add label `kyma-grafana: enabled` along with label `app: <value>` to the **PodTemplate**. Make sure label `app: <value>` is added to either `Deployment` or `Statefulset` spec as well. Then a pre-packaged dashboard with [RED](https://www.weave.works/blog/the-red-method-key-metrics-for-microservices-architecture) and [USE](http://www.brendangregg.com/usemethod.html) metrics for the application gets enabled. The dashboard is defined [here](link).

Reviewer: ```suggestion
- Add the `kyma-grafana: enabled` and the  `app: <value>` label to the **PodTemplate**.  Make sure you add the `app: <value>` label either to `Deployment` or `Statefulset` specification as well. Performing this step enables the pre-packaged dashboard with [RED](https://www.weave.works/blog/the-red-method-key-metrics-for-microservices-architecture) and [USE](http://www.brendangregg.com/usemethod.html) metrics for the application. The dashboard is defined [here](link).
```

Author: done

> - The dashboard is available as `Services` where the application is visible with the value of it's **app** label in its concerned namespace.

Reviewer: ```suggestion
- To see the dashboard, go to  **General > Services** in Grafana. You find the application using the value of its `app` label and the Namespace you specified in in your configuration.
```

Author: done

> - To see the dashboard, go to **General > Services** in Grafana. You find the application using the value of its `app` label and the Namespace you specified in your configuration.

Reviewer: ```suggestion
- To see the dashboard, go to **General > Services** in Grafana. There, you can find your application using the values for the `app` label and the Namespace you specified in your configuration.
```

## Enable alerts

> - Add label `kyma-alerts: enabled` along with label `app: <value>` to your Kyma component. Make sure label `app: <value>` is added to either `Deployment` or `Statefulset` spec as well. Then a pre-packaged sets of alert rules get enabled. The alert rules are defined [here](link).

Reviewer: ```suggestion
- Add the `kyma-alerts: enabled` and the `app: <value>` label to your Kyma component. Make sure you add the `app: <value>` label either to `Deployment` or `Statefulset` specification as well. Performing this step enables pre-packaged sets of alert rules. The alert rules are defined [here](link).
```

Author: done

> ## Sample application where monitoring is enabled

Reviewer: ```suggestion
## Sample application with enables monitoring
```

Author: done

```
apiVersion: extensions/v1beta1
kind: Deployment
metadata:
  labels:
    app: demo-service
  name: demo-service
  namespace: stage
spec:
  replicas: 1
  selector:
    matchLabels:
      app: demo-service
  template:
    metadata:
      labels:
        app: demo-service
        kyma-alerts: enabled
        kyma-grafana: enabled
    spec:
      containers:
      - image: shazra/demo-service
        imagePullPolicy: IfNotPresent
        ports:
        - containerPort: 5000
          protocol: TCP
          name: http
        name: demo-service
        resources:
          limits:
            cpu: 100m
            memory: 256Mi
          requests:
            cpu: 10m
            memory: 128Mi
      restartPolicy: Always
---
apiVersion: v1
kind: Service
metadata:
  labels:
    app: demo-service
  name: demo-service
  namespace: stage
spec:
  ports:
  - name: http
    port: 5000
    protocol: TCP
    targetPort: 5000
  selector:
    app: demo-service
```

> - Dashboard for this application will be available in Grafana `Services` dashboard as `demo-service` in namespace `stage`.

Reviewer: ```suggestion
- The dashboard for this application is available in Grafana under **General > Services**.  To display it, select `demo-service` in the `stage` Namespace.
```

Author: done
